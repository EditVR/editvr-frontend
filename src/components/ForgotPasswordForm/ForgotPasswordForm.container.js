/**
 * @file ForgotPasswordForm.container.js
 * Exports a redux-connected ForgotPasswordForm component.
 */

import { connect } from 'react-redux';

import ForgotPasswordForm from './ForgotPasswordForm';

const mapDispatchToProps = dispatch => ({
  dispatch
});

const mapState = ({ user }) => ({
  user
});

export default connect(mapState, mapDispatchToProps)(ForgotPasswordForm);
