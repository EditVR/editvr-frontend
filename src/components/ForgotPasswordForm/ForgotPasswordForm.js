/**
 * @file ForgotPasswordForm.js
 * Exports a component that allows users to reset their EditVR account password.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { TextField, Button, withStyles } from '@material-ui/core';
import { withFormik } from 'formik';
import { string, object } from 'yup';

import { Message } from '../';
import ForgotPasswordFormStyles from './ForgotPasswordForm.style';
import {
  USER_RESET_PASSWORD,
  FORM_BUTTON_RESET_PASSWORD,
  ERROR_API_USER_EMAIL_NOT_FOUND
} from '../../constants';

const ForgotPasswordForm = ({
  classes,
  user: { error },
  errors,
  touched,
  isSubmitting,
  handleSubmit,
  handleChange,
  handleBlur
}) => (
  <form onSubmit={handleSubmit}>
    {error && <Message type="error">{error}</Message>}
    <TextField
      required
      id="email"
      label="Email"
      type="email"
      className={classes.textField}
      onChange={handleChange}
      onBlur={handleBlur}
      error={!!errors.email && touched.email}
      disabled={isSubmitting}
      helperText={
        errors.email
          ? errors.email
          : 'Enter the email address for your account.'
      }
    />
    <Button
      variant="raised"
      color="primary"
      type="submit"
      className={classes.button}
    >
      {`${FORM_BUTTON_RESET_PASSWORD}`}
    </Button>
  </form>
);

ForgotPasswordForm.propTypes = {
  classes: PropTypes.shape({
    textField: PropTypes.string.isRequired,
    button: PropTypes.string.isRequired
  }).isRequired,
  user: PropTypes.shape({
    error: PropTypes.string
  }),
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleBlur: PropTypes.func.isRequired,
  isSubmitting: PropTypes.bool,
  errors: PropTypes.shape({
    email: PropTypes.oneOfType([PropTypes.string, PropTypes.bool])
  }).isRequired,
  touched: PropTypes.shape({
    email: PropTypes.bool
  }).isRequired
};

ForgotPasswordForm.defaultProps = {
  isSubmitting: false,
  user: {
    error: null
  }
};

const FormikForgotPasswordForm = withFormik({
  displayName: 'ForgotPasswordForm',
  enableReinitialize: true,
  validationSchema: object().shape({
    email: string()
      .email()
      .required()
      .min(3)
      .max(100)
  }),
  handleSubmit: (values, { props, setSubmitting, setErrors }) => {
    const { dispatch } = props;
    const { email } = values;
    dispatch({
      type: USER_RESET_PASSWORD,
      email,
      successHandler: () => {
        // After sending email, tell the user?
        setSubmitting(false);
      },
      errorHandler: error => {
        const message = error.toString();
        if (message.includes(ERROR_API_USER_EMAIL_NOT_FOUND)) {
          setErrors({ email: 'Account matching email not found.' });
        }
        setSubmitting(false);
      }
    });
  }
})(ForgotPasswordForm);

export default withStyles(ForgotPasswordFormStyles)(FormikForgotPasswordForm);
