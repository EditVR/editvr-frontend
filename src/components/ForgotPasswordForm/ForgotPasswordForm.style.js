/**
 * @file ForgotPasswordForm.style.js
 * Exports ForgotPasswordForm component styles.
 */

export default theme => ({
  textField: {
    width: '100%',
    marginTop: `${theme.spacing.unit * 2}px`
  },
  button: {
    marginTop: `${theme.spacing.unit * 3}px`,
    marginRight: theme.spacing.unit
  }
});
