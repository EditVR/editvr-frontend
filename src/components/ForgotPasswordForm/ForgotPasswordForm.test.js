/**
 * @file ForgotPasswordForm.test.js
 * Contains tests for ForgotPasswordForm.js.
 */

import React from 'react';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import ForgotPasswordForm from './ForgotPasswordForm.container';

describe('<ForgotPasswordForm />', () => {
  it('Matches its snapshot', () => {
    const store = configureStore()({
      user: {
        error: null
      }
    });

    expect(
      renderer.create(<ForgotPasswordForm store={store} />).toJSON()
    ).toMatchSnapshot();
  });
});
