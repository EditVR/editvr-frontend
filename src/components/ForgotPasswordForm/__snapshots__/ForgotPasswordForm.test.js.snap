// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPasswordForm /> Matches its snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p
    className="MuiTypography-root-3 MuiTypography-body1-12"
    type="body1"
  >
    To re-set your password, please enter the email associated with your EditVR account in the form below, and click Reset Password. Once you do this, you will be redirected to the Login page, and an email will be sent to your email address with further instructions.
  </p>
  <div
    aria-describedby="email-helper-text"
    className="MuiFormControl-root-27 C-textField-1"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-36 MuiInputLabel-root-31 MuiInputLabel-formControl-32 MuiInputLabel-animated-35"
      data-shrink={false}
      htmlFor="email"
    >
      Email
      <span
        className="MuiFormLabel-asterisk-40"
      >
        â€‰*
      </span>
    </label>
    <div
      className="MuiInput-root-41 MuiInput-formControl-42 MuiInput-underline-45"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-49 MuiInput-inputType-52"
        defaultValue={undefined}
        disabled={false}
        id="email"
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder={undefined}
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="email"
        value={undefined}
      />
    </div>
    <p
      className="MuiFormHelperText-root-54"
      id="email-helper-text"
    >
      Enter the email address for your account.
    </p>
  </div>
  <button
    className="MuiButtonBase-root-74 MuiButton-root-58 MuiButton-raised-64 MuiButton-raisedPrimary-65 C-button-2"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="submit"
  >
    <span
      className="MuiButton-label-59"
    >
      Reset Password
    </span>
    <span
      className="MuiTouchRipple-root-77"
    />
  </button>
</form>
`;
